literals/literals.go

```diff
@@ -1,15 +1,147 @@
-package main
+package literals
 
 import (
 	"fmt"
 	"go/ast"
 	"go/token"
 	"go/types"
+	"log"
+	mathrand "math/rand"
 	"strconv"
 
 	"golang.org/x/tools/go/ast/astutil"
 )
 
+type stringObfuscator interface {
+	obfuscateString(data string) *ast.BlockStmt
+}
+
+type boolObfuscator interface {
+	obfuscateBool(data bool) *ast.BlockStmt
+}
+
+type byteObfuscator interface {
+	obfuscateByte(data []byte) *ast.BlockStmt
```

Comment:
stringObfuscator and byteObfuscator should really be the same. Both can be implemented in terms of `[]byte`, because a string is really just a read-only `[]byte`.

The resulting AST will be slightly different due to the type, but you could just wrap the resulting `[]byte` in a string conversion to obtain a string. Or vice versa; if the shared implementation returns a string, the other can convert to `[]byte`.